app/services/utilization.rb

```diff
@@ -0,0 +1,69 @@
+#
+# This service class calculates shelter utilization
+#
+class Utilization
+  def initialize(params)
+    @group = params[:group_by].parameterize
+    @date = Date.today
+  end
+
+  def calculate
+    case @group
+    when 'shelter'
+      averages_by_building
+    when 'case-type'
+      averages_by_case_type
+    else
+      raise ArgumentError, 'Invalid group_by parameter'
+    end
+  end
+
+  private
+
+  def utilizations
+    Census.group(:shelter_building_id).average(:count)
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  def averages_by_building
+    ShelterBuilding.all.map do |sb|
+      capacity = sb.places.count
+      utilization = utilizations[sb.id]
+      record = {
+        facility: sb.shelter.name,
```

Comment:
It looks like there is a domain class trying to emerge here and below.
The domain class could then just use json serialization and the method would no longer be too long.
Can you refactor out a domain class 
it could be called shelterUtilization